### PR TITLE
Update nikic/php-parser to 4.15.1

### LINKED ISSUE #7062: AnnotationToAttributeRector breaks code formatting
In my rector config I have:

```php
$services = $containerConfigurator->services();
$services
    ->set(AnnotationToAttributeRector::class)
    ->configure([new AnnotationToAttribute('Symfony\Component\Routing\Annotation\Route')]);
```

When I run it, rector breaks my code formatting:
obraz

Blank lines are removed and whitespace is added before `:` in return type.

### PR DESCRIPTION
**Pull request by samsonasik**
nikic/php-parser has new release 4.15.0 https://github.com/nikic/PHP-Parser/releases/tag/v4.15.0 which fix method spacing, ref:

- https://github.com/nikic/PHP-Parser/pull/880

Fixes https://github.com/rectorphp/rector/issues/7062

### CODE REVIEW

**Comment by samsonasik** [comment]
it seems cause error on AnnotationToAttributeRector tests:

```diff
There were 3 failures:

1) Rector\Tests\Php80\Rector\Class_\AnnotationToAttributeRector\AnnotationToAttributeRectorTest::test with data set #4 ('/Users/samsonasik/www/rector-...hp.inc')
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 
 final class TrailingComma
 {
-    #[GenericAnnotation(key: 'value')]
-    #[GenericAnnotation(some: 'item', summary: 'item')]
+    #[GenericAnnotation(key: 'value')]#[GenericAnnotation(some: 'item', summary: 'item')]
     protected $someColumn;
 }
```

**Comment by TomasVotruba** [comment]
I'll see what needs to be done

**Comment by TomasVotruba** [comment]
I asked for the regression.
See https://github.com/nikic/PHP-Parser/commit/a2608f0b742b711d24de914d41c0162e37494ea6#commitcomment-82995459

**Comment by samsonasik** [comment]
@TomasVotruba I tried create PR https://github.com/nikic/PHP-Parser/pull/881 to fix it.

**Comment by samsonasik** [comment]
New release 4.15.1 https://github.com/nikic/PHP-Parser/releases/tag/v4.15.1 , Let's update https://github.com/rectorphp/rector-src/pull/2903/commits/aeb358e935a714b2781ff3b4501f69bc945e56b8

**Comment by samsonasik** [comment]
It seems still error on rector-doctrine:

```diff
There was 1 failure:

1) Rector\Doctrine\Tests\Set\DoctrineORM29Set\DoctrineORM29SetTest::test with data set #7 ('/home/runner/work/rector-src/...hp.inc')
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 #[Entity]
 class User
 {
-    #[ManyToOne(targetEntity: 'HelloWorld\UserGroup', inversedBy: 'users')]
+     #[ManyToOne(targetEntity: 'HelloWorld\UserGroup', inversedBy: 'users')]
     #[JoinColumn(name: 'group_id', referencedColumnName: 'id')]
```

and rector-downgrade-php:

```bash
 There were 4 errors:

1) Rector\Tests\DowngradePhp71\Rector\ConstFetch\DowngradePhp71JsonConstRector\DowngradePhp71JsonConstRectorTest::test with data set #3 ('/home/runner/work/rector-src/...hp.inc')
PHPStan\BetterReflection\Identifier\Exception\InvalidIdentifierName: Invalid identifier name "0"
```

**Comment by samsonasik** [comment]
Ok, on rector-doctrine error, it seems due to spacing on start comment on annotation first:

```diff
diff --git a/tests/Set/DoctrineORM29Set/Fixture/join_column_many_to_one.php.inc b/tests/Set/DoctrineORM29Set/Fixture/join_column_many_to_one.php.inc
index e7134b7..3c3a578 100644
--- a/tests/Set/DoctrineORM29Set/Fixture/join_column_many_to_one.php.inc
+++ b/tests/Set/DoctrineORM29Set/Fixture/join_column_many_to_one.php.inc
@@ -13,7 +13,7 @@ use Doctrine\ORM\Mapping\JoinColumn;
  */
 class User
 {
-     /**
+    /**
```

so the AttributeGroup generation start from that.

**Comment by samsonasik** [comment]
https://github.com/rectorphp/rector-doctrine/pull/134

**Comment by samsonasik** [comment]
I added handling `Invalid identifier name "0"` error for rector-downgrade-php tests https://github.com/rectorphp/rector-src/pull/2903/commits/9ebec21c221c9241f5440909622554a58291908d

**Comment by samsonasik** [comment]
@TomasVotruba to make CI green, https://github.com/rectorphp/rector-doctrine/pull/134 need to be merged.

**Comment by TomasVotruba** [comment]
Thank you, this is very helpful upgrade :+1:

File: packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php [hunk 1]
```diff
@@ -319,6 +320,24 @@ private function processTernary(Ternary $ternary, MutatingScope $mutatingScope):
...
+     */
+    private function processNodesFromNodeScopeResolver(
+        array $stmts,
+        MutatingScope $mutatingScope,
+        callable $nodeCallback
+    ): void {
+        try {
+            $this->nodeScopeResolver->processNodes($stmts, $mutatingScope, $nodeCallback);
+        } catch (InvalidIdentifierName $invalidIdentifierName) {
+            if ($invalidIdentifierName->getMessage() !== 'Invalid identifier name "0"') {
```
Comment (TomasVotruba) [inline packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php]: I think this should be fixed propperly. What is the root of this issue?

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php]: That's due to ConstFetch with Name("0"), I created new PR https://github.com/rectorphp/rector-downgrade-php/pull/24 for it

Comment (samsonasik) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php]: I reverted the tweak https://github.com/rectorphp/rector-src/pull/2903/commits/14cbabfcff7c3ce4cb7f5d8f53d774b57d73c96a 

Now to make CI green,  https://github.com/rectorphp/rector-downgrade-php/pull/24 need to be merged first.

Comment (TomasVotruba) [reply packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php]: :partying_face:

**Comment by TomasVotruba** [comment]
Thank you :clap:

**Pull request merged**